lib/runtime-core/src/codegen.rs

```diff
@@ -137,22 +137,22 @@ impl<
     }
 }
 
-pub fn default_validating_parser_config() -> wasmparser::ValidatingParserConfig {
+pub fn validating_parser_config(features: &Features) -> wasmparser::ValidatingParserConfig {
     wasmparser::ValidatingParserConfig {
         operator_config: wasmparser::OperatorValidatorConfig {
             enable_threads: false,
             enable_reference_types: false,
-            enable_simd: true,
+            enable_simd: features.simd,
             enable_bulk_memory: false,
             enable_multi_value: false,
         },
         mutable_global_imports: false,
     }
 }
 
-fn validate(bytes: &[u8]) -> CompileResult<()> {
+fn validate(bytes: &[u8], features: &Features) -> CompileResult<()> {
```

Comment:
Add a new method, rather than `validate` use `validate_with_features`, this way we will not break compatibility with the API.

Reply:
I think this one isn't part of the exported API, the public one is `pub fn validate` in `lib.rs`? Should I avoid breaking compatibility here anyways?

Reply:
I think you are right on the API exposure (since validate was not public).
However I still see some utility into having two functions `validate` and `validate_with_feature`. Thoughts?

Reply:
Renamed to `validate_with_features` and removed dead function `validate`.